File: leetcode/solutions/0476.NumberComplement/findComplement.go
package solution

import "math"

func findComplement(num int) int {
  if num == 0 {
    return 0
  }

  numOfDigits := math.Floor(math.Log2(float64(num))) + 1

  return num ^ int(math.Pow(2, numOfDigits) - 1)
}

File: leetcode/solutions/0476.NumberComplement/findComplement.ts
export default function findComplement(num: number): number {
  if (num === 0) {
    return 0;
  }

  const numOfDigits = Math.floor(Math.log2(num)) + 1;

  return num ^ (Math.pow(2, numOfDigits) - 1);
}
